tests/functional/test_integration.py

```diff
@@ -43,27 +67,39 @@ async def test_build_and_deploy_with_psql(
         check=True,
     )
     await ops_test.model.wait_for_idle(status=ActiveStatus.name, raise_on_error=False)
-    subprocess.check_output(
-        ["juju", "destroy-model", influx_model_name, "--force", "--destroy-storage", "-y"]
+    await ops_test.juju(
+        "destroy-model",
+        f"{controller_name}:{influx_model_name}",
+        "--force",
+        "--destroy-storage",
+        "-y",
     )
-    subprocess.check_output(["juju", "switch", "k8s-ctrl"])
 
 
 async def test_status(ops_test: OpsTest):
+    """
+    arrange: given the resulting juju model of the first test
+    assert: the model should have all its charms in active state.
```

Comment:
What about `act`? E.g., something like, "when the status of the applications is checked"